std/experimental/color/conv.d

```diff
@@ -0,0 +1,523 @@
+
+            // convert and return the output
+            static if(To.hasAlpha)
+                return To(convertPixelType!ToType(r), convertPixelType!ToType(g), convertPixelType!ToType(b), convertPixelType!ToType(src[3]));
+            else
+                return To(convertPixelType!ToType(r), convertPixelType!ToType(g), convertPixelType!ToType(b));
+        }
+    }
+    else static if(isRGB!From && isXYZ!To)
+    {
+        alias ToType = To.ComponentType;
+        alias FromType = From.ComponentType;
+        alias WorkType = WorkingType!(FromType, ToType);
+
+        // unpack the working values
+        auto src = color.tristimulus;
+        WorkType r = convertPixelType!WorkType(src[0]);
+        WorkType g = convertPixelType!WorkType(src[1]);
+        WorkType b = convertPixelType!WorkType(src[2]);
+
+        static if(From.linear == false)
+        {
+            r = toLinear!(From.colorSpace)(r);
+            g = toLinear!(From.colorSpace)(g);
+            b = toLinear!(From.colorSpace)(b);
+        }
+
+        // transform to XYZ
+        enum toXYZ = RGBColorSpaceMatrix!(From.colorSpace, WorkType);
+        WorkType[3] v = multiply(toXYZ, [r, g, b]);
+        return To(v[0], v[1], v[2]);
+    }
+    else static if(isXYZ!From && isRGB!To)
+    {
+        alias ToType = To.ComponentType;
+        alias FromType = From.ComponentType;
+        alias WorkType = WorkingType!(FromType, ToType);
+
+        enum toRGB = inverse(RGBColorSpaceMatrix!(To.colorSpace, WorkType));
+        WorkType[3] v = multiply(toRGB, [ WorkType(color.X), WorkType(color.Y), WorkType(color.Z) ]);
+
+        static if(To.linear == false)
+        {
+            v[0] = toGamma!(To.colorSpace)(v[0]);
+            v[1] = toGamma!(To.colorSpace)(v[1]);
+            v[2] = toGamma!(To.colorSpace)(v[2]);
+        }
+
+        return To(convertPixelType!ToType(v[0]), convertPixelType!ToType(v[1]), convertPixelType!ToType(v[2]));
+    }
+
+    // *** xyY triplet ***
+    else static if(isxyY!From && isxyY!To)
+    {
+        alias F = To.ComponentType;
+        return To(F(color.x), F(color.y), F(color.Y));
+    }
+    else static if(isxyY!From && isXYZ!To)
+    {
+        alias F = To.ComponentType;
+        if(color.y == F(0))
+            return To(F(0), F(0), F(0));
+        else
+            return To(F(color.x*color.Y/color.y), F(color.Y), F((F(1)-color.x-color.y)*color.Y/color.y));
+    }
+    else static if(isXYZ!From && isxyY!To)
+    {
+        alias F = To.ComponentType;
+        auto sum = color.X + color.Y + color.Z;
+        if(sum == F(0))
+            return To(WhitePoint!F.D65.x, WhitePoint!F.D65.y, F(0));
+        else
+            return To(F(color.X/sum), F(color.Y/sum), F(color.Y));
+    }
+
+    // *** fallback plan ***
+    else
+    {
+        // cast along a conversion path to reach our target conversion
+        alias Path = ConversionPath!(From, To);
+        return convertColor!To(convertColor!(Path[0])(color));
+    }
+}
+
+unittest
```

Comment:
Yeah, the idea is that the example code you would normally put in the documentation would now be a unit test with documentation attached to itself. This way the example code will be executed when the unit tests are run, forcing the example code to always be up to date and working.